spec/traces/backend/open_telemetry_spec.rb

```diff
@@ -29,21 +28,96 @@ def my_method(argument)
 
 Traces::Provider(MyClass) do
 	def my_method(argument)
-		trace('my_method', attributes: {argument: argument}) {super}
+		Traces.trace('my_method', attributes: {argument: argument}) {super}
+	end
+	
+	def my_span
+		Traces.trace('my_span') {|span| return span}
+	end
+
+	def my_context
+		Traces.trace('my_context') {|span| return Traces.trace_context}
+	end
+
+	def my_span_and_context
+		Traces.trace('my_span_and_context') {|span| return span, Traces.trace_context}
 	end
 end
 
 RSpec.describe Traces::Backend::OpenTelemetry do
+	subject(:instance) { MyClass.new }
+	
 	it "has a version number" do
 		expect(Traces::Backend::OpenTelemetry::VERSION).not_to be nil
 	end
 	
 	it "can invoke trace wrapper" do
-		instance = MyClass.new
+		expect(Traces).to receive(:trace).and_call_original
+		instance.my_method(10)
+	end
+	
+	describe OpenTelemetry::Trace::Span do
+		it "can yield an open telemetry span" do
+			expect(Traces).to receive(:trace).and_call_original
+			expect(instance.my_span).to be_a(described_class)
+		end
+	end
+	
+	describe "span_and_context" do
+		let(:span_and_context) {instance.my_span_and_context}
+		let(:span) {span_and_context.first}
+		let(:context) {span_and_context.last}
+		let(:new_context) {
+			Traces::Context.new(
+				SecureRandom.uuid,
+				SecureRandom.uuid,
+				0,
+				nil,
+				remote: false
+			)
+		}
 		
-		expect(instance).to receive(:trace).and_call_original
-		expect(Traces::Backend::OpenTelemetry::TRACER).to receive(:start_span).and_call_original
+		it "can provide a trace context with a trace_id" do
+			expect(context).to have_attributes(trace_id: String)
+			expect(span.context.trace_id).to eq(context.trace_id)
+		end
 		
-		instance.my_method(10)
+		describe "#trace_context=" do
+			it "can update the trace context" do
+				expect(Traces.trace_context).not_to eq(new_context)
+				
+				Traces.trace_context = new_context
+				expect(instance.my_span.context).to have_attributes(
+					trace_id: new_context.trace_id,
+					span_id: new_context.parent_id
+				)
+			end
+			
+			it "can roundtrip the trace context" do
+				parsed_context = Traces::Context.parse(context.to_s)
+
+				Traces.trace_context = parsed_context
+
+				expect(instance.my_span.context).to have_attributes(
+					trace_id: context.trace_id,
+					span_id: context.parent_id
+				)
+			end
+			
+			it "can update trace context within active span" do
+				
+				Traces.trace("test.outer") do |span|
+					Traces.trace_context = new_context
+					
+					# It's updated on the next call to trace:
+					expect(span.context.trace_id).not_to eq(new_context.trace_id)
```

Comment:
ditto on this spec, its failing, but I think my assertions are correct assuming i understand the feature supported.